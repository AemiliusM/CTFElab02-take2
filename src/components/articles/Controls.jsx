import React from 'react';

export default function Controls({
    country,
    onCountryChange,
    onSubmit,}) 
    {
    return (
        <form onSubmit={onSubmit}>
        <label htmlFor='country'>Country</label>
        <input 
        id='country'
        name='country'
        type='text'
        value={country}
        onChange={onCountryChange}
        />
        <button aria-label='get-articles'>Select from Country</button>
        </form>
    )
}