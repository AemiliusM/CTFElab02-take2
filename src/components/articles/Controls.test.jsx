import React, { Component } from 'react';
import { render } from '@testing-library/react';
import Controls from './Controls';

describe('Query test', () => {
    it('takes in query and confirms state change', () => {
        const { asFragment } = render(
            <Controls 
                onSubmit='ca' value='ca' onCountryChange='ca'
                />
        );
        expect(asFragment()).toMatchSnapshot();
    });
});