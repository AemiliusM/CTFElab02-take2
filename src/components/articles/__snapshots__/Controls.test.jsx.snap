// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Query test takes in query and confirms state change 1`] = `
<DocumentFragment>
  <form>
    <label
      for="country"
    >
      Country
    </label>
    <input
      id="country"
      name="country"
      type="text"
      value=""
    />
    <button
      aria-label="get-articles"
    >
      Select from Country
    </button>
  </form>
</DocumentFragment>
`;
